Update verus to support rust 1.88 and under

ziqiaozhou opened the pull request:
* Update operator usage with the latest external trait features.
* Distinguish usize vs u64 proofs due to https://github.com/verus-lang/verus/pull/1745
* Repair deprecated verus grammars
* This should close https://github.com/coconut-svsm/svsm/issues/700

ziqiaozhou commented:
To run verus, users need to re-run the `script/vinstall.sh` to update all verus related tools and need to install 1.88 rust and then run  `cargo verify`. So the current verus version should be compatible with svsm project until svsm decides to update to >1.88

stefano-garzarella commented:
@ziqiaozhou please add `Closes #700` in the commit message when @luigix25 will confirm it, so the issue will be automatically closed.

stefano-garzarella commented on kernel/src/address.rs:
```diff
@@ -280,10 +280,6 @@ impl ops::Sub<PhysAddr> for PhysAddr {
     type Output = InnerAddr;
 
     #[inline]
-    #[verus_spec(ret =>
```
Is this removing intentional?

stefano-garzarella replied:
Okay, I see. Next time please describe it the commit message.

luigix25 commented:
@stefano-garzarella yes, I had to manually install the rust toolchain too.
I wonder why `scripts/vinstall.sh` worked fine on your system without the override in place. Maybe we should modify that script?

stefano-garzarella commented:
> @stefano-garzarella yes, I had to manually install the rust toolchain too. I wonder why `scripts/vinstall.sh` worked fine on your system without the override in place. 

I guess because initially I run what you suggested, so it installed the right toolchain, but IMO we should document that.

> Maybe we should modify that script?

Or document in `Documentation/docs/developer/VERIFICATION.m` that prerequisite if verus can't be built with 1.86

ziqiaozhou commented:
> > @stefano-garzarella yes, I had to manually install the rust toolchain too. I wonder why `scripts/vinstall.sh` worked fine on your system without the override in place.
> 
> I guess because initially I run what you suggested, so it installed the right toolchain, but IMO we should document that.
> 
> > Maybe we should modify that script?
> 
> Or document in `Documentation/docs/developer/VERIFICATION.m` that prerequisite if verus can't be built with 1.86

I updated the script and added the reason:

> Verus requires a specific version of Rust (e.g., 1.88.0) because it depends
  on the compiler’s internal libraries for verification. Thanks to Rust’s strong
  backward compatibility guarantees, Verus can analyze code written against
  older versions. However, some newer Verus features depend on recent rustc
  versions, so we may occasionally upgrade the toolchain to support those
  features, which does not guarantee it is the exact version defined by
  `rust-toolchain.toml`

stefano-garzarella commented:
Or maybe we can just set and export RUSTUP_TOOLCHAIN:


```
$ cat scripts/vinstall.sh
#!/bin/bash
# SPDX-License-Identifier: MIT OR Apache-2.0
#
# Copyright (c) Microsoft Corporation
#
# Author: Ziqiao Zhou <ziqiaozhou@microsoft.com>
# A script to install verus tools
VERISMO_REV=7545c6c
export RUSTUP_TOOLCHAIN="1.87.0"
RUSTUP_TARGET="x86_64-unknown-none"

# Verus is based on 1.88.0, so we need to install the toolchain first.
if ! rustup toolchain list | grep "$RUSTUP_TOOLCHAIN"; then
  rustup install
fi
if ! rustup target list --installed | grep "$RUSTUP_TARGET"; then
  rustup target add "$RUSTUP_TARGET"
fi

# Install verus tools
cargo install --git https://github.com/microsoft/verismo/ --rev $VERISMO_REV cargo-v
builtin=`cargo metadata --format-version 1 | jq -r '.packages[] | select(.name == "builtin_macros") | .targets[].src_path'`
verus=`dirname $builtin`/../../../source/target-verus/release/verus
if [ -f ${verus} ]; then
    echo "verus (${verus}) is already built"
else
    # build the verus using the source code from builtin/vstd defined in
    # Cargo.toml so that the verus lib and tool are compatible.
   cargo v prepare-verus
fi

# verus-rustc as a wrapper to call verus with proper rustc flags.
cargo install --git https://github.com/microsoft/verismo/ --rev $VERISMO_REV verus-rustc

# verus formatter
cargo install --git https://github.com/verus-lang/verusfmt --rev v0.5.7
```

ziqiaozhou commented:
Thanks for trying out and suggest the improved scripts. I updated based on your suggestion and further improved it by getting RUSTUP_TOOLCHAIN from verus toolchain file.

stefano-garzarella commented:
@ziqiaozhou okay, now it's much better, but please squash the last commit with the previous one. I don't think make sense to introduce code in a commit and remove it in the next commit of the same PR.

nit: I'd suggest following the stile of the other commit messages (e.g. avoid bulleted lists)
